include/openssl/evp_kdf.h

```diff
@@ -70,6 +70,9 @@ int EVP_KDF_derive(EVP_KDF_CTX *ctx, unsigned char *key, size_t *keylen);
 # define EVP_KDF_CTX_set_md(kctx, md) \
             EVP_KDF_CTX_ctrl(kctx, EVP_KDF_CTRL_MD, 0, (void *)(md))
 
+# define EVP_KDF_CTX_set1_key(kctx, key, keylen) \
```

Comment:
Sorry for the possibly stupid question but why set1 instead of just set ?
I may be missing the context of what the 1 means here...

Reply:
This is a common OpenSSL idiom. We should really write this down somewhere in the documentation!

get0 == get a value such that the pointer points to the internal data value. The caller should not free it
get1 == get a value such that the pointer is to memory that the caller should free
set0 == set a value and pass the ownership of the pointer to OpenSSL. Once successfully set the caller should not directly free it
set1 == set a value but retain ownership of the pointer with the caller, i.e. the caller should free it

Reply:
I like to think of it as "0 (or 1) references/copies are created"